lib/fetchers/fetcher.py

```diff
@@ -66,57 +68,18 @@ def remove_blacklisted_sets(self):
                 if set in self.sets:
                     del self.sets[set]
 
-    def request_content_from(self, url, params={}, attempts=3):
+    def request_content_from(self, url, params={}):
         error = None
         resp = None
-        m = re.match(r"(https?)://(.*?)(/.*)", url)
-        socket_parts = m.group(2).split(":")
-        host = socket_parts[0]
-        ssl = m.group(1) == "https"
-
-        if len(socket_parts) == 2:
-            port = int(socket_parts[1])
-        elif ssl:
-            port = 443
-        else:
-            port = 80
-
-        req_uri = m.group(3)
-        if params:
-            if "?" in req_uri:
-                req_uri += "&" + urlencode(params, True)
-            else:
-                req_uri += "?" + urlencode(params, True)
-
-        for i in range(attempts):
-            try:
-                if ssl:
-                    con = httplib.HTTPSConnection(host, port)
-                else:
-                    con = httplib.HTTPConnection(host, port)
-                con.request("GET", req_uri, None, self.http_headers)
-                resp = con.getresponse()
-                if resp.status == 200:
-                    break  # connection stays open for read()!
-                con.close()
-            except Exception as e:
-                if type(e) == socket.error:
-                    err_msg = e
-                else:
-                    err_msg = e.message
-                print >> sys.stderr, "Requesting %s: %s" % (req_uri, err_msg)
-            time.sleep(2)
-
-        if resp and resp.status == 200:
-            rv = resp.read()
-            con.close()
-        elif resp:
-            error = "Error ('%d') requesting %s" % (resp.status, url)
-            rv = None
+
+        r = requests.get(url, params=params, headers=self.http_headers)
+
+        if r.status_code == 200:
```

Comment:
That's actually a Grace Hopper quotation: https://en.wikiquote.org/wiki/Grace_Hopper#Quotes